src/test/ttfi_pipeline.ml

```diff
@@ -17,121 +142,222 @@ module Pipeline_1 (Bfx : Biokepi.EDSL.Semantics) = struct
     |> Bfx.list
 
 
-  let mutect_on_fastqs ~reference_build ~normal ~tumor =
-    let aligner =
-      Bfx.lambda (fun fq -> Bfx.bwa_aln ~reference_build fq) in
-    let align_list list_of_fastqs =
-      Bfx.list_map list_of_fastqs ~f:aligner |> Bfx.merge_bams
+  let every_vc_on_fastqs ~reference_build ~normal ~tumor =
+    let aligner which_one =
+      Bfx.lambda (fun fq -> which_one ~reference_build fq) in
+    let align_list how (list_of_fastqs : [ `Fastq ] list Bfx.repr) =
+      Bfx.list_map list_of_fastqs ~f:how |> Bfx.merge_bams
+    in
+    let aligners = Bfx.list [
+        aligner @@ Bfx.bwa_aln ?configuration: None;
+        aligner @@ Bfx.bwa_mem ?configuration: None;
+        aligner @@ Bfx.hisat ~configuration:Biokepi.Tools.Hisat.Configuration.default_v1;
+        aligner @@ Bfx.hisat ~configuration:Biokepi.Tools.Hisat.Configuration.default_v2;
+        aligner @@ Bfx.star ~configuration:Biokepi.Tools.Star.Configuration.Align.default;
+        aligner @@ Bfx.mosaik;
+        aligner @@ (fun ~reference_build fastq ->
+            Bfx.bwa_aln ~reference_build fastq
+            |> Bfx.bam_to_fastq ~sample_name:"realigned" `PE
+            |> Bfx.bwa_mem ?configuration: None ~reference_build
+          );
+      ] in
+    let somatic_of_pair how =
+      Bfx.lambda (fun pair ->
+          let normal = Bfx.pair_first pair in
+          let tumor = Bfx.pair_second pair in
+          how ~normal ~tumor)
+    in
+    let somatic_vcs =
+      List.map ~f:somatic_of_pair [
+        Bfx.mutect
+          ~configuration:Biokepi.Tools.Mutect.Configuration.default;
+        Bfx.mutect2
+          ~configuration:Biokepi.Tools.Gatk.Configuration.Mutect2.default;
+        Bfx.somaticsniper
+          ~configuration:Biokepi.Tools.Somaticsniper.Configuration.default;
+        Bfx.strelka
+          ~configuration:Biokepi.Tools.Strelka.Configuration.default;
+        Bfx.varscan_somatic ?adjust_mapq:None;
+        Bfx.muse
+          ~configuration:Biokepi.Tools.Muse.Configuration.wes;
+        Bfx.virmid
+          ~configuration:Biokepi.Tools.Virmid.Configuration.default;
+      ]
+    in
+    let aligned_pairs
+      : (([ `Fastq ] list * [ `Fastq ] list) -> ([ `Bam ] * [ `Bam ]) list) Bfx.repr =
+      Bfx.lambda (fun pair ->
+          Bfx.list_map aligners ~f:(
+            Bfx.lambda (fun (al : ([ `Fastq ] -> [ `Bam ]) Bfx.repr) ->
+                Bfx.pair
+                  (align_list al (Bfx.pair_first pair : [ `Fastq ] list Bfx.repr))
+                  (align_list al (Bfx.pair_second pair))
+              )
+          )
+        )
+    in
+    let vcfs =
```

Comment:
This kind of thing is what seems a bit boilerplatey (though I guess unavoidable) The main thing going on here is you're lifting OCaml lambdas (and HOFs) to TTFI lambdas so that the HOFs/etc here can be accessed by the type system/compiler, right?